morebits.js

```diff
@@ -2625,6 +2601,45 @@ Morebits.wiki.page = function(pageName, currentAction) {
 		}
 	};
 
+	/**
+	 * Marks the page as patrolled, using rcid (if available) or revid
+	 *
+	 * Patrolling as such doesn't need to rely on loading the page in
+	 * question; simply passing a revid to the API is sufficient, so in
+	 * those cases just using Morebits.wiki.api is probably preferable.
+	 *
+	 * No error handling since we don't actually care about the errors
+	 */
+	this.patrol = function() {
+		if (!Morebits.userIsSysop && !Morebits.userIsInGroup('patroller')) {
+			return;
+		}
+
+		// If patrolling current page, don't need to query for latest revID
+		if ($('.patrollink').length) {
+			var patrolhref = $('.patrollink a').attr('href');
+			ctx.rcid = mw.util.getParamValue('rcid', patrolhref);
+			fnProcessPatrol(this, this);
+		} else if (Morebits.string.toUpperCaseFirstChar(mw.config.get('wgPageName')).replace(/ /g, '_').trim() ===
+			Morebits.string.toUpperCaseFirstChar(ctx.pageName).replace(/ /g, '_').trim()) {
```

Comment:
#915 adds `mw.Title` dependency to morebits. Might as well use that here.

Reply:
When I get a chance, I want to see what adding some of the `mw.Title` features into `morebits.wiki.page` for this sort of thing would look like.  Seems like it'd be useful, didn't want to embark on something *else* new for this, but, yes.

Reply:
I meant changing the "poor man's normalisation" to`new mw.Title(Morebits.pageNameNorm).toText() === new mw.Title(ctx.pageName).toText()`. Regarding the larger question of integrating mw.Title with morebits.wiki.page, what would probably make sense is to make morebits.wiki.page a sub-class of mw.Title (rather than trying to duplicate mw.Title code).

Reply:
Sorry, I was thinking out loud, orthogonal to this, about some advantages to having it around.  Something like (albeit nicer, with checks, etc.) could be helpful:

```js
    this.getTalkPageName = function() {
        return mw.Title.newFromText(this.getPageName()).getTalkPage().getPrefixedText();
    };
    this.getSubjectPageName = function() {
        return mw.Title.newFromText(this.getPageName()).getSubjectPage().getPrefixedText();
    };
```

As for here, yeah, sure.  I suppose it's simple enough to include in something off-topic.

Reply:
Done here.  Without testing I'm not entirely sure what the `.trim()` was doing, I guess `ctx.pageName` would have whitespace?